homeassistant/components/freebox/router.py

```diff
@@ -0,0 +1,199 @@
+        self.mac = None
+        self._sw_v = None
+        self._attrs = {}
+
+        self.devices: Dict[str, any] = {}
+        self.sensors: Dict[str, any] = {}
+
+        self.listeners = []
+
+    async def setup(self) -> None:
+        """Set up a Freebox router."""
+        self._api = await get_api(self.hass, self._host)
+
+        try:
+            await self._api.open(self._host, self._port)
+        except HttpRequestError:
+            _LOGGER.exception("Failed to connect to Freebox")
+            return ConfigEntryNotReady
+
+        # System
+        fbx_config = await self._api.system.get_config()
+        self.mac = fbx_config["mac"]
+        self._name = fbx_config["model_info"]["pretty_name"]
+        self._sw_v = fbx_config["firmware_version"]
+
+        # Devices & sensors
+        await self.update_all()
+        async_track_time_interval(self.hass, self.update_all, SCAN_INTERVAL)
+
+    async def update_all(self, now: Optional[datetime] = None) -> None:
+        """Update all Freebox platforms."""
+        await self.update_sensors()
+        await self.update_devices()
+
+    async def update_devices(self) -> None:
+        """Update Freebox devices."""
+        new_device = False
+        fbx_devices: Dict[str, any] = await self._api.lan.get_hosts_list()
+
+        # Adds the Freebox itself
+        fbx_devices.append(
+            {
+                "primary_name": self._name,
+                "l2ident": {"id": self.mac},
+                "vendor_name": "Freebox SAS",
+                "host_type": "router",
+                "active": True,
+                "attrs": self._attrs,
+            }
+        )
+
+        for fbx_device in fbx_devices:
+            device_mac = fbx_device["l2ident"]["id"]
+
+            if self.devices.get(device_mac) is None:
+                new_device = True
+
+            self.devices[device_mac] = fbx_device
+
+        async_dispatcher_send(self.hass, self.signal_device_update)
+
+        if new_device:
+            async_dispatcher_send(self.hass, self.signal_device_new)
+
+    async def update_sensors(self) -> None:
+        """Update Freebox sensors."""
+        # System sensors
+        syst_datas: Dict[str, any] = await self._api.system.get_config()
+
+        # According to the doc `syst_datas["sensors"]` is temperature sensors in celsius degree.
+        # Name and id of sensors may vary under Freebox devices.
+        for sensor in syst_datas["sensors"]:
+            self.sensors[sensor["id"]] = {
+                **TEMPERATURE_SENSOR_TEMPLATE,
+                **{
+                    SENSOR_NAME: f"Freebox {sensor['name']}",
+                    SENSOR_VALUE: sensor["value"],
+                },
+            }
+
+        # Connection sensors
+        connection_datas: Dict[str, any] = await self._api.connection.get_status()
+        for sensor_key in CONNECTION_SENSORS:
+            self.sensors[sensor_key] = {
+                **CONNECTION_SENSORS[sensor_key],
```

Comment:
As above. We should separate the entity details from the router device data.

Reply:
As above